internal/controller/nonadminbackup_controller.go

```diff
@@ -70,167 +62,46 @@ const (
-		//       be reverted back to the previous state - the state which created VeleroBackup
-		//       in a first place.
-		logger.Info("Backup already exists, updating NonAdminBackup status", nameField, veleroBackupName)
-		updatedNab, errBackupUpdate := function.UpdateNonAdminBackupFromVeleroBackup(ctx, r.Client, logger, &nab, &veleroBackup)
-		// Regardless if the status was updated or not, we should not
-		// requeue here as it was only status update.
-		if errBackupUpdate != nil {
-			return ctrl.Result{}, errBackupUpdate
-		} else if updatedNab {
-			logger.V(1).Info("NonAdminBackup CR - Rqueue after Status Update")
-			return ctrl.Result{Requeue: true, RequeueAfter: requeueTimeSeconds * time.Second}, nil
-		}
-		return ctrl.Result{}, nil
 	}
 
-	// Ensure labels are set for the Backup object
-	existingLabels := veleroBackup.Labels
-	naManagedLabels := function.AddNonAdminLabels(existingLabels)
-	veleroBackup.Labels = naManagedLabels
-
-	// Ensure annotations are set for the Backup object
-	existingAnnotations := veleroBackup.Annotations
-	ownerUUID := string(nab.ObjectMeta.UID)
-	nabManagedAnnotations := function.AddNonAdminBackupAnnotations(nab.Namespace, nab.Name, ownerUUID, existingAnnotations)
-	veleroBackup.Annotations = nabManagedAnnotations
-
-	_, err = controllerutil.CreateOrPatch(ctx, r.Client, &veleroBackup, nil)
-	if err != nil {
-		logger.Error(err, "Failed to create backup", nameField, veleroBackupName)
-		return ctrl.Result{}, err
-	}
-	logger.Info("VeleroBackup successfully created", nameField, veleroBackupName)
-
-	// Phase: Created
-	// BackupAccepted: True # do not update
-	// BackupQueued: True
-	_, errUpdate = function.UpdateNonAdminPhase(ctx, r.Client, logger, &nab, nacv1alpha1.NonAdminBackupPhaseCreated)
-	if errUpdate != nil {
-		logger.Error(errUpdate, "Unable to set NonAdminBackup Phase: Created", nameField, req.Name, constant.NameSpaceString, req.Namespace)
-		return ctrl.Result{}, errUpdate
-	}
-	_, errUpdate = function.UpdateNonAdminBackupCondition(ctx, r.Client, logger, &nab, nacv1alpha1.NonAdminConditionAccepted, metav1.ConditionTrue, "Validated", "Valid Backup config")
-	if errUpdate != nil {
-		logger.Error(errUpdate, "Unable to set BackupAccepted Condition: True", nameField, req.Name, constant.NameSpaceString, req.Namespace)
-		return ctrl.Result{}, errUpdate
-	}
-	_, errUpdate = function.UpdateNonAdminBackupCondition(ctx, r.Client, logger, &nab, nacv1alpha1.NonAdminConditionQueued, metav1.ConditionTrue, "BackupScheduled", "Created Velero Backup object")
-	if errUpdate != nil {
-		logger.Error(errUpdate, "Unable to set BackupQueued Condition: True", nameField, req.Name, constant.NameSpaceString, req.Namespace)
-		return ctrl.Result{}, errUpdate
-	}
+	// Run Reconcile Batch as we have the NonAdminBackup object
+	reconcileExit, reconcileRequeue, reconcileErr := ReconcileBatch(
+		ReconcileFunc(func(...any) (bool, bool, error) {
```

Comment:
all these new functions have same params (ctx, log, nab, right :question: I would enforcing typing here

Reply:
They may not have the same in the future, we will be adding different batch for Restore etc... and I want this to be common function.

Reply:
Current implementation is very different without ReconcileBatch so we don't have that anymore.